### PR TITLE
Handle IME via sane keyboard util

### PR DESCRIPTION
**Pull request by yukipastelcat**
Ticket: https://oat-sa.atlassian.net/browse/INF-543

## What's Changed

- Added Japanese IME composition handling to saneKeyboardEvents (compositionstart / compositionupdate / compositionend, composing input).
- Suppressed romaji leakage during IME by skipping keydown / typedText while composing (isComposing, keyCode === 229).
- Synced interim kana into MathQuill via updateCompositionText / syncCompositionText, with guards for:
- empty textarea (skip-empty)
- unchanged composition string (skip-unchanged)
- empty compositionend when interim text is already shown
- Fixed programmatic composition clear to use controller.backspace() (with noop keyboard-event fallback) instead of keystroke('Backspace', null), which crashed on e.preventDefault().
- Kept iPad Japanese IME fallback via keydown.key when the textarea stays empty.
- Added temporary console.debug diagnostics prefixed with [Math Entry PCI | MathQuill] for delivery troubleshooting.
- Extended unit tests for IME composition, iPad fallback, growing composition, and regression cases.

## TODO 

- [x] Unit tests

## How to test

1. Build mathquill: `make`
2. Run unit tests: `make test`
3. Open test/unit.html and confirm the saneKeyboardEvents IME suites pass.
4. Integrate into Math Entry PCI:
    - Copy build/mathquill.js into extension-tao-itemqti-pci/.../runtime/mathquill/mathquill.js
    - Rebuild PCI (grunt portableelement -e=qtiItemPci -i=mathEntryInteraction) and upload the PCI zip to any environment.
5. Manual IME check (macOS / iPad, Japanese input):
    - Open a Math Entry item in test delivery
    - Type hiragana (e.g. わをんー) - interim kana should update live, no romaji leakage, no erase after first character
    - Type English math (e.g. x+1) — should still work
    - Optional debug: in browser DevTools, enable Verbose and filter console by Math Entry PCI to inspect composition event flow (sync, skip-empty, skip-unchanged, no preventDefault errors).

## Summary by CodeRabbit

* **New Features**
  * Added improved IME composition tracking with real-time synchronization of interim text (including Japanese IME).
  * Enhanced composition lifecycle handling: typing is suppressed during composition and committed text is applied on completion.
* **Bug Fixes**
  * Prevented duplicate kana insertion on iPad/Japanese IME flows.
  * Avoided unintended clearing of interim composition text and refined backspacing behavior during updates.
* **Tests**
  * Added extensive IME and iPad/Japanese IME coverage (start/update/end, empty event edge cases, sync behavior).
  * Added a `deepEqual` test assertion helper.

### CODE REVIEW

**Review by KirylHatalski — APPROVED** [review]

File: src/services/saneKeyboardEvents.util.ts [hunk 1]
```diff
@@ -125,8 +125,21 @@ var saneKeyboardEvents = (function () {
...
+    var isComposing = false;
+    var compositionTextLength = 0;
+    var compositionString = '';
     var is_iPad = isIpadOS();
 
+    var LOG_PREFIX = '[Math Entry PCI | MathQuill]';
+    var noopKeyboardEvent = {
+      preventDefault: noop,
+    } as unknown as KeyboardEvent;
+    function debugIme(event: string, detail?: object) {
```
Comment (Karol-Stelmaczonek) [inline src/services/saneKeyboardEvents.util.ts]: I don't think we need debug functions on production code, lets remove it please.

**Review by Karol-Stelmaczonek — APPROVED** [review]

File: src/services/saneKeyboardEvents.util.ts (hunk 1 above)
Comment (yukipastelcat) [reply src/services/saneKeyboardEvents.util.ts]: Removed in [6c8b3dc](https://github.com/oat-sa/mathquill/pull/14/commits/6c8b3dc4037e9206b5d96185e3a0d5bea92047df)

File: test/unit/saneKeyboardEvents.test.js [hunk 2]
```diff
@@ -506,4 +506,237 @@ suite('saneKeyboardEvents', function () {
...
+      assert.deepEqual(typed, ['か']);
+      el.val('かな');
+      el.trigger('compositionupdate');
+      assert.equal(counter, 3, 'updated kana shown during composition');
+      assert.deepEqual(typed, ['か', 'な']);
+      el.val('かな');
+      el.trigger('compositionend');
+      assert.equal(counter, 3, 'no duplicate insert on composition end');
+      assert.equal(el.val(), '', 'textarea cleared after commit');
+    });
```
[the remark this answers is not in the document]
Comment (yukipastelcat) [reply test/unit/saneKeyboardEvents.test.js:633]: Check fixes in [e80efee](https://github.com/oat-sa/mathquill/pull/14/commits/e80efee7954a1baa5e0e43f77186d6e64b60d1f1)
